mu-plugins/rest-api/endpoints/class-wporg-rest-users-controller.php

```diff
@@ -58,6 +58,59 @@ public function register_routes() {
 				'schema' => array( $this, 'get_public_item_schema' ),
 			)
 		);
+
+		// Filter the post endpoints to alter the public URL for non-authenticated users.
+		$post_types = get_post_types(
+			array(
+				'public'       => true,
+				'show_in_rest' => true,
+			)
+		);
+
+		foreach ( $post_types as $post_type ) {
+			add_filter( "rest_prepare_{$post_type}", array( $this, 'alter_post_author_link' ) );
+		}
+	}
+
+	/**
+	 * Change the author link for posts to this endpoint.
+	 * 
+	 * @param \WP_REST_Response $response Response object for the request.
+	 * 
+	 * @return \WP_REST_Response Response object for the request.
+	 */
+	public function alter_post_author_link( $response ) {
+		$links = $response->get_links();
+		if (
+			empty( $links['author'][0]['href'] ) ||
+			! preg_match( '!wp/v2/users/(?P<id>\d+)!i', $links['author'][0]['href'], $m )
+		) {
+			return $response;
+		}
+
+		$user_id = absint( $m['id'] );
```

Comment:
In retrospect, this may be cleaner...

```suggestion
			add_filter( "rest_prepare_{$post_type}", array( $this, 'alter_post_author_link' ), 10, 2 );
		}
	}

	/**
	 * Change the author link for posts to this endpoint.
	 * 
	 * @param \WP_REST_Response $response Response object for the request.
	 * 
	 * @return \WP_REST_Response Response object for the request.
	 */
	public function alter_post_author_link( $response, $post ) {
		$links = $response->get_links();
		if (
			empty( $links['author'][0]['href'] ) ||
			! str_contains( $links['author'][0]['href'], 'wp/v2/users/' ) ||
			! $post->post_author
		) {
			return $response;
		}

		$user_id = $post->post_author;
```

Reply:
It might be even simpler to just check if the response has an `author` property. Something like:

```php
$data = $response->get_data();
$user = get_user_by( 'id', $data['author'] ?? 0 );
if ( ! $user->exists() ) {
	return $response;
}
```